module/documents/item.mjs

```diff
@@ -1837,15 +1837,19 @@ export default class Item5e extends SystemDocumentMixin(Item) {
    * @param {object} [options]
    * @param {boolean} [options.deterministic] Whether to force deterministic values for data properties that could be
    *                                          either a die term or a flat term.
+   * @param {boolean} [options.extended]      Include extra values like "name" that aren't useful in formulas.
    */
-  getRollData({ deterministic=false }={}) {
+  getRollData({ deterministic=false, extended=false }={}) {
     let data;
-    if ( this.system.getRollData ) data = this.system.getRollData({ deterministic });
+    if ( this.system.getRollData ) data = this.system.getRollData({ deterministic, extended });
     else {
       if ( !this.actor ) return null;
-      data = { ...this.actor.getRollData({ deterministic }), item: { ...this.system } };
+      data = { ...this.actor.getRollData({ deterministic, extended }), item: { ...this.system } };
+    }
+    if ( data?.item ) {
+      data.item.flags = { ...this.flags };
+      if ( extended ) data.item.name = this.name;
```

Comment:
Do we want to include some logic for unidentified items here?

```suggestion
      if ( extended ) data.item.name = game.user.isGM || (this.system.identified !== false) ? this.name : this.system.unidentified.name;
```

Reply:
The name is already swapped during data preparation to use the unidentified name, so this is handled automatically (unless we wanted to handle this differently and have it always show the normal name to GMs, currently we show the unidentified name to everyone if it is unidentified).

Reply:
Oh that's right, I had forgotten about that. Maybe always showing the GM the real name is desirable, I don't really know, so I guess we just leave it for now.